### PR TITLE
Migrate the Customers > Title (Only Delete & Bulk Delete)

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Migrate the Customers > Title (Only Delete & Bulk Delete)
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Cf #10628 
| How to test?      | Make assets and Go to this URI: `/configure/shop/title`<br />You should be able to delete & bulk delete

### CODE REVIEW

**Comment by Progi1984** [comment]
Ping @PrestaShop/prestashop-maintainers for review

File: src/Core/Domain/Title/ValueObject/TitleId.php
Comment (atomiix) [inline src/Core/Domain/Title/ValueObject/TitleId.php]: I think the titleId should be >= 0
```suggestion
        if (0 >= $value) {
```

File: tests/Integration/Behaviour/Features/Context/Domain/TitleFeatureContext.php
Comment (atomiix) [inline tests/Integration/Behaviour/Features/Context/Domain/TitleFeatureContext.php:26]:
```suggestion

declare(strict_types=1);
```

File: tests/Integration/Behaviour/Features/Context/TitleFeatureContext.php
Comment (atomiix) [inline tests/Integration/Behaviour/Features/Context/TitleFeatureContext.php:26]:
```suggestion

declare(strict_types=1);
```

File: tests/Integration/Behaviour/Features/Scenario/Title/title_management.feature
Comment (atomiix) [inline tests/Integration/Behaviour/Features/Scenario/Title/title_management.feature:28]: Why this part is commented?

File: tests/Integration/Behaviour/Features/Scenario/Title/title_management.feature
Comment (atomiix) [inline tests/Integration/Behaviour/Features/Scenario/Title/title_management.feature]:
```suggestion
  Background:
    Given I define an uncreated title "titleNotFound"

  Scenario: Adding new title
```

File: tests/Integration/Behaviour/Features/Context/TitleFeatureContext.php
Comment (atomiix) [inline tests/Integration/Behaviour/Features/Context/TitleFeatureContext.php]:
```suggestion
     * @When I edit title :titleReference with the following properties:
```

File: tests/Integration/Behaviour/Features/Context/TitleFeatureContext.php
Comment (atomiix) [inline tests/Integration/Behaviour/Features/Context/TitleFeatureContext.php]:
```suggestion
     * @When I add a new title :titleReference with the following properties:
```

File: src/Core/Domain/Title/ValueObject/TitleId.php
Comment (Progi1984) [reply src/Core/Domain/Title/ValueObject/TitleId.php]: Fixed

Comment (Progi1984) [reply src/Core/Domain/Title/ValueObject/TitleId.php]: And tested.

File: tests/Integration/Behaviour/Features/Context/Domain/TitleFeatureContext.php
Comment (Progi1984) [reply tests/Integration/Behaviour/Features/Context/Domain/TitleFeatureContext.php:26]: Fixed

File: tests/Integration/Behaviour/Features/Context/TitleFeatureContext.php
Comment (Progi1984) [reply tests/Integration/Behaviour/Features/Context/TitleFeatureContext.php:26]: Fixed

File: tests/Integration/Behaviour/Features/Scenario/Title/title_management.feature
Comment (Progi1984) [reply tests/Integration/Behaviour/Features/Scenario/Title/title_management.feature:28]: It will do in the next PR after this one.

File: tests/Integration/Behaviour/Features/Scenario/Title/title_management.feature
Comment (Progi1984) [reply tests/Integration/Behaviour/Features/Scenario/Title/title_management.feature]: Fixed

File: tests/Integration/Behaviour/Features/Context/TitleFeatureContext.php
Comment (Progi1984) [reply tests/Integration/Behaviour/Features/Context/TitleFeatureContext.php]: Fixed

File: tests/Integration/Behaviour/Features/Context/TitleFeatureContext.php
Comment (Progi1984) [reply tests/Integration/Behaviour/Features/Context/TitleFeatureContext.php]: Fixed

File: src/Core/Domain/Title/Exception/DeleteTitleException.php
Comment (atomiix) [inline src/Core/Domain/Title/Exception/DeleteTitleException.php]: Why this error message is different from the not-bulk one?

File: src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/TitleController.php
Comment (atomiix) [inline src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/TitleController.php]: This argument doesn't seem to be used
```suggestion
    private function getErrorMessages(): array
```

File: src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/TitleController.php
Comment (atomiix) [inline src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/TitleController.php]:
```suggestion
            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
```

File: src/Core/Domain/Title/Exception/DeleteTitleException.php
Comment (Progi1984) [reply src/Core/Domain/Title/Exception/DeleteTitleException.php]: Fixed.

File: src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/TitleController.php
Comment (Progi1984) [reply src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/TitleController.php]: Fixed

File: src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/TitleController.php
Comment (Progi1984) [reply src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/TitleController.php]: Fixed

**Review by atomiix — DISMISSED** [review]

**Review by atomiix — DISMISSED** [review]

**Review by matthieu-rolland — DISMISSED** [review]

**Review by khouloudbelguith — COMMENTED** [review]
Hi @Progi1984,

I have an issue following these steps:

1. Go to BO > Shop Parameters > Customer Settings > Titles tab
2. Resize the menu
3. Delete a title & confirm the deletion
4. See error: An invalid delete button is displayed

image

Using a mobile, try to do the same steps => see error
image

I attached a screen record

https://user-images.githubusercontent.com/16067358/183687877-1d03ad8a-99b2-4051-b233-15bb8e932614.mp4

Thanks!

**Comment by Progi1984** [comment]
@khouloudbelguith Thanks. I fixed your bug. But I need a new review from @PrestaShop/prestashop-maintainers & @PrestaShop/committers.

:notebook: The wording has not changed. But the label, yes.

File: src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/TitleController.php
Comment (zuk3975) [inline src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/TitleController.php]:
```suggestion
     * @return int[]
```

File: src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/TitleController.php
Comment (zuk3975) [inline src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/TitleController.php]: could define array structure with psalm notation

**Review by zuk3975 — DISMISSED** [review]

File: src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/TitleController.php
Comment (Progi1984) [reply src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/TitleController.php]: Fixed

File: src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/TitleController.php
Comment (Progi1984) [reply src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/TitleController.php]: Fixed

**Review by zuk3975 — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Review by khouloudbelguith — COMMENTED** [review]
Hi @Progi1984,

I checked this PR.
Functional tests are ok
But automatic tests are NOK
https://github.com/khouloudbelguith/testing_pr/actions/runs/3135429053/jobs/5092379844
- The test failed "functional:BO:customer-service" not related to this PR (reproduced with develop also)
- The test failed "functional:FO" not related to this PR (reproduced with develop also)

But the test "functional:BO:orders" is reproduced locally also with PR only 
[check_invoice.pdf](https://github.com/PrestaShop/PrestaShop/files/9658155/check_invoice.pdf)

I tried to run this test `HEADLESS=false TEST_PATH="functional/BO/02_orders/01_orders/viewAndEditOrder/11_checkInvoice.js" URL_FO="http://shop.com/" PASSWD="prestashop_demo" npm run test:specific`

Without PR => OK
image

Thanks!

**Review by khouloudbelguith — APPROVED** [review]
Hi @Progi1984,

I checked today the nightly 
https://nightly.prestashop.com/report/5138
It is not related to this PR.

It is QA :heavy_check_mark: 

Thanks!

**Pull request merged**
